orangecontrib/educational/widgets/owgradientdescent.py

```diff
@@ -238,16 +250,46 @@ def __init__(self):
         self.alpha_spin = gui.spin(
             widget=self.properties_box, master=self, callback=self.change_alpha,
             value="alpha", label="Learning rate: ",
-            minv=0.001, maxv=100, step=0.001, spinType=float, decimals=3,
+            minv=1e-5, maxv=1e50, step=0.001, spinType=float, decimals=3,
             alignment=Qt.AlignRight, controlWidth=80)
+        self.step_size_spin = gui.spin(
+            widget=self.properties_box, master=self, callback=self.change_step,
+            value="step_size", label="Num. samples: ",
+            minv=1, maxv=100, step=1, alignment=Qt.AlignRight, controlWidth=80)
         self.stochastic_checkbox = gui.checkBox(
             widget=self.properties_box, master=self,
             callback=self.change_stochastic, value="stochastic",
             label="Stochastic ")
-        self.step_size_spin = gui.spin(
-            widget=self.properties_box, master=self, callback=self.change_step,
-            value="step_size", label="Step size: ",
-            minv=1, maxv=100, step=1, alignment=Qt.AlignRight, controlWidth=80)
+
+        self.combobox_opt = gui.comboBox(
+            widget=self.properties_box, master=self, value="opt_type",
+            label="SGD optimizer: ", items=self.opt_names,
+            orientation=Qt.Horizontal, addSpace=4, callback=self._opt_changed)
+
+        _m_comp = gui.doubleSpin(
+            widget=self.properties_box, master=self, value="momentum",
+            minv=1e-4, maxv=1e+4, step=1e-4, label="Momentum:", decimals=4,
+            alignment=Qt.AlignRight, controlWidth=90,
+            callback=self._opt_changed)
+
+        _r_comp = gui.doubleSpin(
+            widget=self.properties_box, master=self, value="rho", minv=1e-4,
+            maxv=1e+4, step=1e-4, label="Rho:", decimals=4,
+            alignment=Qt.AlignRight, controlWidth=90, callback=self._opt_changed)
+
+        _b1_comp = gui.doubleSpin(
+            widget=self.properties_box, master=self, value="beta1", minv=1e-5,
+            maxv=1e+5, step=1e-4, label="Beta 1:", decimals=5,
```

Comment:
It's supposed to be an educational widget, yet nobody knows what these parameters mean. Perhaps add some `tooltip='...'`?